pymilo/transporters/tree_transporter.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+"""PyMilo SGDOptimizer object transporter."""
+from sklearn.tree._tree import Tree
+
+from .transporter import AbstractTransporter
+from .general_data_structure_transporter import GeneralDataStructureTransporter
+from ..pymilo_param import NUMPY_TYPE_DICT
+
+import numpy as np
+import platform
+
+class TreeTransporter(AbstractTransporter):
+    """Customized PyMilo Transporter developed to handle (pyi,pyx) Tree object."""
+
+    def serialize(self, data, key, model_type):
+        """
+        Serialize instances of the Tree class.
+
+        Record the n_features, n_classes and n_outputs fields of tree object.
+
+        :param data: the internal data dictionary of the given model
+        :type data: dict
+        :param key: the special key of the data param, which we're going to serialize its value(data[key])
+        :type key: object
+        :param model_type: the model type of the ML model
+        :type model_type: str
+        :return: pymilo serialized output of data[key]
+        """
+        if isinstance(data[key], Tree):
+            gdst = GeneralDataStructureTransporter()
+            tree = data[key]
+            tree_inner_state = tree.__getstate__()
+
+            data[key] = {
+                'params': {
+                    'internal_state': {
+                        "max_depth": tree_inner_state["max_depth"],
+                        "node_count": tree_inner_state["node_count"],
+                        "nodes": {
+                            "types": [str(np.dtype(i).name) for i in tree_inner_state["nodes"][0]],
+                            "field-names": list(tree_inner_state["nodes"][0].dtype.names),
+                            "values": [node.tolist() for node in tree_inner_state["nodes"]],
+                            },
+                        "values":  gdst.ndarray_to_list(tree_inner_state["values"]),
+                    },
+                    'n_features': tree.n_features,
+                    'n_classes': gdst.ndarray_to_list(tree.n_classes),
+                    'n_outputs': tree.n_outputs,
+                }
+            }
+
+        return data[key]
+
+    def deserialize(self, data, key, model_type):
+        """
+        Deserialize the special tree_ field of the Decision Trees.
+
+        The associated tree_ field of the pymilo serialized model, is extracted through
+        it's previously serialized parameters.
+
+        deserialize the data[key] of the given model which type is model_type.
+        basically in order to fully deserialize a model, we should traverse over all the keys of its serialized data dictionary and
+        pass it through the chain of associated transporters to get fully deserialized.
+
+        :param data: the internal data dictionary of the associated json file of the ML model which is generated previously by
```

Comment:
Please summarize it in a single line. My suggestion is: 
"the internal data dictionary of the associated JSON file of the ML model generated by pymilo export"